src/proposed/0140-budget-limit-threshold.md

```diff
@@ -0,0 +1,469 @@
+<!--
```

Comment:
I'm coming to this PR very late, since it's being reviewed by the TSC this week.  I'm not vehemently opposed to the change but I'm not yet convinced by the motivation and would love to see my questions answered. It's great to see work happening on the MCS api and I love to see it develop and evolve.